libraries/SRV_Channel/SRV_Channel.cpp

```diff
@@ -170,6 +170,20 @@ float SRV_Channel::get_output_norm(void)
     return ret;
 }
 
+float SRV_Channel::get_output_norm_from_pwm(uint16_t pwm)
+{
+    float ret;
+    if (pwm < servo_trim) {
+        ret = (float)(pwm - servo_trim) / (float)(servo_trim - servo_min);
+    } else if (pwm > servo_trim) {
+        ret = (float)(pwm - servo_trim) / (float)(servo_max  - servo_trim);
+    } else {
+        ret = 0;
+    }
+    
```

Comment:
Just for 4. (since the others were already agreed upon on Gitter): how do you guarantee that all callers of this method are passing a PWM value obtained from RCOutput? You are just thinking of your code, but once this method is in, it can be used by anyone. Like I said, having a *do_reverse* argument to the method is a good idea, if you want to have it disabled by default, that's fine.